gadget/install/install.go

```diff
@@ -160,8 +160,13 @@ func Run(model gadget.Model, gadgetRoot, device string, options Options, observe
 			logger.Noticef("encrypted device %v", part.Node)
 		}
 
-		if err := makeFilesystem(&part); err != nil {
-			return nil, err
+		// use the diskLayout.SectorSize here instead of lv.SectorSize, we check
+		// that if there is a sector-size specified in the gadget that it
+		// matches what is on the disk, but sometimes there may not be a sector
+		// size specified in the gadget.yaml, but we will always have the sector
+		// size from the physical disk device
+		if err := makeFilesystem(&part, diskLayout.SectorSize); err != nil {
```

Comment:
I had an issue with this when testing on a real device a gadget with unset sector-size and non-512 sector size, we should add another/more tests about that case from this level too